test/jdk/java/net/HttpURLConnection/HttpURLConnectionExpectContinueTest.java

```diff
@@ -0,0 +1,398 @@
+                            if (control.write100ContinueTwice) {
+                                outputStream.write("HTTP/1.1 100 Continue\r\n\r\n".getBytes());
+                                outputStream.flush();
+                            }
+                        }
+
+                        StringBuilder contentLengthBuilder = new StringBuilder();
+                        while (true) {
+                            b = (byte) inputStreamReader.read();
+                            contentLengthBuilder.append((char) b);
+
+                            if (contentLengthBuilder.length() >= 2) {
+                                char[] lastBytes = new char[2];
+                                contentLengthBuilder.getChars(
+                                        contentLengthBuilder.length() - 2,
+                                        contentLengthBuilder.length(), lastBytes, 0);
+                                if (Arrays.equals(lastBytes, new char[]{'\r', '\n'})) {
+                                    String lengthInHex =
+                                            contentLengthBuilder.substring(0, contentLengthBuilder.length() - 2);
+
+                                    int contentLength = Integer.parseInt(lengthInHex, 16);
+                                    char[] body = new char[contentLength];
+                                    inputStreamReader.read(body);
+                                    break;
+                                    //normally we have to parse more data,
+                                    //but for simplicity we expect no more chunks...
+                                }
+                            }
+                        }
+                    }
+                    outputStream.write(control.response.getBytes());
+                    outputStream.flush();
+                } catch (SocketException e) {
+                    //ignore
+                } catch (IOException e) {
+                    throw new RuntimeException(e);
+                }
+            }
+        };
+        serverThread = new Thread(runnable);
+        serverThread.start();
+    }
+
+    @AfterAll
+    public void stopServerSocket() throws Exception {
+        control.stop = true;
+        control.serverSocket.close();
+        serverThread.join();
+    }
+
+    @Test
+    public void testNonChunkedRequestWithExpect100ContinueResponse() throws Exception {
+        URL url = new URL("http://localhost:" + control.serverSocket.getLocalPort());
+
+        String body = "testNonChunkedRequestWithExpect100ContinueResponse";
+        control.response = "HTTP/1.1 200 OK\r\n" +
+                "Connection: close\r\n" +
+                "Content-Length: " + body.length() + "\r\n" +
+                "\r\n" +
+                body + "\r\n";
+
+        control.respondWith100Continue = true;
+        control.write100ContinueTwice = false;
+
+        HttpURLConnection connection = (HttpURLConnection) url.openConnection();
```

Comment:
Hi Darragh,

It looks like the tests share a lot of similar code, unless there is a reason not to, could they not share a similar function to reduce code duplication?

Reply:
Hey, thanks for pointing that out, I'll try and reduce some of the duplication